### PR TITLE
[batch] miscellaneous usability improvements

### PR DESCRIPTION
**Pull request by jigold**
- provide a default session
- made cancel not wait for all jobs to be cancelled before returning
- some log statements
- fix error catching in retry function
- add retry to batch client create jobs
- add cancel if batch is not submitted successfully

### CODE REVIEW

**Comment by johnc1231** [comment]
You have some failing tests right now. I'll review once you fix?

File: hail/python/hailtop/batch_client/aioclient.py [hunk 1]
```diff
@@ -342,6 +345,18 @@ def create_job(self, image, command=None, args=None, env=None, ports=None,
...
+    async def _submit_job_with_retry(self, batch_id, docs):
+        n_attempts = 0
+        saved_err = None
+        while n_attempts < max_job_submit_attempts:
+            try:
+                return await self._client._post(f'/api/v1alpha/batches/{batch_id}/jobs/create', json={'jobs': docs})
+            except Exception as err:  # pylint: disable=W0703
+                saved_err = err
+                n_attempts += 1
+                await asyncio.sleep(0.5)
```
Comment (danking) [inline hail/python/hailtop/batch_client/aioclient.py]: check which ones have already been made then send the diff

Comment (jigold) [reply hail/python/hailtop/batch_client/aioclient.py]: This isn't necessary. None are committed to the database if any in the batch fail.

**Comment by danking** [comment]
Looks like some test are failing for real though.

**Pull request merged**